add the variables of batch-dml (#10872)
pingcap/docs: system-variables.md

    In the case of a poor network environment, appropriately increasing the value of this variable can effectively alleviate error reporting to the application end caused by timeout. If the application end wants to receive the error information more quickly, minimize the value of this variable.

> ### tidb_batch_commit
> - Scope: Session
> - Persists to cluster: No
> - Type: Boolean
> - Default value: `OFF`
> - The variable is used to control whether to enable the deprecated batch commit feature. When truned on, a transaction may be split into multiple transactions by grouping every a few statements and be committed non-atomically. It is not recommended to turn on this feature.

Reviewer: ```suggestion
### tidb_batch_commit

> **Warning:**
>
> It is **NOT** recommended to enable this variable.

- Scope: Session
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`
- The variable is used to control whether to enable the deprecated batch-commit feature. When this variable is enabled, a transaction might be split into multiple transactions by grouping a few statements and committed non-atomically, which is not recommended.
```

> ### tidb_batch_delete
> - Scope: Session
> - Persists to cluster: No
> - Type: Boolean
> - Default value: `OFF`
> - This variable is used to control whether to enable the batch delete feature, which is part of the deprecated batch-dml feature. When truned on, delete statements may be split into multiple transactions and be committed non-atomically. You must also turn on `tidb_enable_batch_dml` and set a positive `tidb_dml_batch_size` to make it work.
> 
> > **Warning:**
> > batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.

Reviewer: ```suggestion
### tidb_batch_delete

> **Warning:**
>
> This variable is associated with the deprecated batch-dml feature, which might cause data corruption. Therefore, it is not recommended to enable this variable for batch-dml. Instead, use [non-transactional DML](/non-transactional-dml.md).

- Scope: Session
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`
- This variable is used to control whether to enable the batch-delete feature, which is a part of the deprecated batch-dml feature. When this variable is enabled, `DELETE` statements might be split into multiple transactions and committed non-atomically. To make it work, you also need to enable `tidb_enable_batch_dml` and set a positive value for `tidb_dml_batch_size`, which is not recommended.
```

### tidb_batch_insert
- Scope: Session
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`
- This variable is used to control whether to enable the batch insert feature, which is part of the deprecated batch-dml feature. When truned on, insert statements may be split into multiple transactions and be committed non-atomically. You must also turn on `tidb_enable_batch_dml` and set a positive `tidb_dml_batch_size` to make it work.

> **Warning:**

> > batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.

Author: batch-insert cannot be replaced by non-transactional dml for now, but will be.

> ### tidb_batch_insert
> - Scope: Session
> - Persists to cluster: No
> - Type: Boolean
> - Default value: `OFF`
> - This variable is used to control whether to enable the batch insert feature, which is part of the deprecated batch-dml feature. When truned on, insert statements may be split into multiple transactions and be committed non-atomically. You must also turn on `tidb_enable_batch_dml` and set a positive `tidb_dml_batch_size` to make it work.
> 
> > **Warning:**
> > batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.

Reviewer: ```suggestion
### tidb_batch_insert

> **Warning:**
>
> This variable is associated with the deprecated batch-dml feature, which might cause data corruption. Therefore, it is not recommended to enable this variable for batch-dml. Instead, use [non-transactional DML](/non-transactional-dml.md).

- Scope: Session
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`
- This variable is used to control whether to enable the batch-insert feature, which is a part of the deprecated batch-dml feature. When this variable is enabled, `INSERT` statements might be split into multiple transactions and committed non-atomically. To make it work, you also need to enable `tidb_enable_batch_dml` and set a positive for `tidb_dml_batch_size`, which is not recommended.
```

- Unit: Rows
- When this value is greater than `0`, TiDB will batch commit statements such as `INSERT` or `LOAD DATA` into smaller transactions. This reduces memory usage and helps ensure that the `txn-total-size-limit` is not reached by bulk modifications.
- Only the value `0` provides ACID compliance. Setting this to any other value will break the atomicity and isolation guarantees of TiDB.

> - You must also turn on `tidb_enable_batch_dml` and at least one of `tidb_batch_insert` and `tidb_batch_delete` to use this feature.

Reviewer: ```suggestion
- To make this variable work, you also need to enable `tidb_enable_batch_dml` and at least one of `tidb_batch_insert` and `tidb_batch_delete`.
```

> 
> > **Warning:**
> > batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.

Reviewer: It is recommended to move this warning to line 1109.
```suggestion
```

- Default value: `OFF`
- This variable is used to determine whether to include the `AUTO_INCREMENT` columns when creating a generated column or an expression index.

> ### tidb_enable_batch_dml
> - Scope: GLOBAL
> - Persists to cluster: Yes
> - Type: Boolean
> - Default values: `OFF`
> - This variable contronls whether to enable the deprecated batch-dml feature. When it's enabled, certain statements may be split into multiple transactions, which means it's non-atomic and should be used with care. When using the feature you must ensure there is no concurrent opertaions on the data you are operating on. You must also specify a positive `tidb_batch_dml_size` and turn on at least one of `tidb_batch_insert` and `tidb_batch_delete` to make it work.
> 
> > **Warning:**
> > batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.

Reviewer: ```suggestion
### tidb_enable_batch_dml

> **Warning:**
>
> This variable is associated with the deprecated batch-dml feature, which might cause data corruption. Therefore, it is not recommended to enable this variable for batch-dml. Instead, use [non-transactional DML](/non-transactional-dml.md).

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Boolean
- Default values: `OFF`
- This variable controls whether to enable the deprecated batch-dml feature. When it is enabled, certain statements might be split into multiple transactions, which is non-atomic and should be used with care. When using batch-dml, you must ensure that there are no concurrent operations on the data you are operating on. To make it work, you must also specify a positive value for `tidb_batch_dml_size` and enable at least one of `tidb_batch_insert` and `tidb_batch_delete`.
```
